client/my-sites/plugins/with-plugin-redirect.js

```diff
@@ -0,0 +1,183 @@
+ * @returns {object|boolean} Plugin redirect object. Otherwise, False.
+ */
+
+function getPluginRedirect( slug ) {
+	if ( ! redirectingPluginsList.hasOwnProperty( slug ) ) {
+		return false;
+	}
+
+	return redirectingPluginsList[ slug ];
+}
+
+const withPluginRedirect = createHigherOrderComponent(
+	( Component ) => ( props ) => {
+		const { plugin } = props;
+
+		if ( ! plugin?.slug ) {
+			return <Component { ...props } />;
+		}
+
+		const pluginSlug = plugin.slug;
+		const redirectPlugin = getPluginRedirect( pluginSlug );
+		if ( ! redirectPlugin ) {
+			return <Component { ...props } />;
+		}
+
+		const wasInstalling = useRef();
+		const [ redirectTo, setRedirectTo ] = useState( false );
+
+		const translate = useTranslate();
+		const dispatch = useDispatch();
+
+		const {
+			hasSiteBeenTransferred,
+			hasPluginBeenInstalled,
+			isInstallingPlugin,
+			isJetpack,
+			isAtomic,
+			isSiteConnected,
+			redirectUrl,
+		} = useSelector(
+			( state ) => {
+				const siteId = getSelectedSiteId( state );
+				const transferState = getAutomatedTransferStatus( state, siteId );
+
+				const isInstalling = isPluginActionInProgress( state, siteId, pluginSlug, INSTALL_PLUGIN );
+				const hasBeenInstalled = isPluginActionCompleted(
+					state,
+					siteId,
+					pluginSlug,
+					INSTALL_PLUGIN
+				);
+
+				return {
+					hasSiteBeenTransferred: transferState === transferStates?.COMPLETE,
+					isInstallingPlugin: isInstalling,
+					hasPluginBeenInstalled: hasBeenInstalled,
+					isAtomic: isSiteAutomatedTransfer( state, siteId ),
+					isJetpack: isJetpackSite( state, siteId ),
+					isSiteConnected: getSiteConnectionStatus( state, siteId ),
+					redirectUrl: redirectPlugin.handler( state, siteId ),
+				};
+			},
+			[ pluginSlug ]
+		);
+
+		useEffect( () => {
+			if ( ! isSiteConnected ) {
+				return;
+			}
+
+			// Store the previous state of `isInstalling`.
+			if ( isJetpack && isInstallingPlugin ) {
+				wasInstalling.current = true;
+			}
+
+			// Store the previous state of `isInstalling`.
+			if ( isAtomic && hasSiteBeenTransferred ) {
+				wasInstalling.current = true;
+			}
+
+			if (
+				( isJetpack && ! isInstallingPlugin && wasInstalling.current ) || // <- Jetpack site.
+				( isAtomic && ! hasSiteBeenTransferred && wasInstalling.current ) // <- Atomic site.
+			) {
+				setRedirectTo( redirectUrl );
+			}
+		}, [
+			isInstallingPlugin,
+			hasPluginBeenInstalled,
+			redirectUrl,
+			hasSiteBeenTransferred,
+			isJetpack,
+			isAtomic,
+			isSiteConnected,
+		] );
+
+		useEffect( () => {
+			if ( ! redirectTo ) {
+				return;
+			}
+
+			// For Atomic sites, we shoulnd't wait since
+			/// there is not a Notice.
+			const pluginInstalledNoticeTime = hasSiteBeenTransferred ? 0 : 3000;
+
+			let timerId = setTimeout( () => {
+				dispatch( removeNotice( 'plugin-notice' ) );
+
+				// re-use same timer ID to redirect the client.
+				timerId = setTimeout( () => {
+					window.location.href = redirectTo;
+				}, 5500 );
```

Comment:
Would be great if we add the `5000` to a `noticeDuration` variable, and then here we have `noticeDuration + 500`. That would definitely make it more clear about the intent and where this number is coming from.

Reply:
oh, sorry I opted to don't close the notification so the user will see it until it redirects